### DISCUSSION TITLE
Array and base_column nesting in migrations and create_pydantic_model

### DISCUSSION THREAD

**Discussion by williamflaherty** [piccolo-orm/piccolo#809] [Q&A] [answered]
Auto migrations don't appear to capture changes to base_column type in an Array column, which can cause discrepancies between the database and the tables. This can be solved with manual migrations but that may be something worth documenting? Or is there potentially a deeper issue? 

Additionally, create_pydantic_model does not look at the configuration of the base_column (null=True, default =None, etc) so when validating data does not properly take into account what's acceptable/expected. Any plans to add that?

**Comment by dantownsend** [comment, marked as answer]
Thanks - they are some good observations which need fixing.

> Auto migrations don't appear to capture changes to base_column type in an Array column, which can cause discrepancies between the database and the tables. This can be solved with manual migrations but that may be something worth documenting? Or is there potentially a deeper issue?

Good point - the migrations might not be handling that correctly. Array columns are a special case which require specific logic. The migrations should distinguish between different array types. Hopefully this is a fairly simple fix. I'll create an issue.

> Additionally, create_pydantic_model does not look at the configuration of the base_column (null=True, default =None, etc) so when validating data does not properly take into account what's acceptable/expected. Any plans to add that?

I hadn't considered that before. This is something we should be able to change quite easily.

**Reply by williamflaherty** [reply]
Great news thanks!
